frame/transaction-payment/src/lib.rs

```diff
@@ -906,4 +975,35 @@ mod tests {
 			assert_eq!(System::events().len(), 0);
 		});
 	}
+
+	#[test]
+	fn refund_consistent_with_actual_weight() {
+		ExtBuilder::default()
+			.balance_factor(10)
+			.base_weight(7)
+			.build()
+			.execute_with(||
+		{
+			let info = info_from_weight(100);
+			let post_info = post_info_from_weight(33);
+			let prev_balance = Balances::free_balance(2);
+			let len = 10;
+			let tip = 5;
+
+			NextFeeMultiplier::put(Fixed128::saturating_from_rational(1, 4));
+
+			let pre = ChargeTransactionPayment::<Runtime>::from(tip)
+				.pre_dispatch(&2, CALL, &info, len)
+				.unwrap();
+
+			ChargeTransactionPayment::<Runtime>
+				::post_dispatch(pre, &info, &post_info, len, &Ok(()))
+				.unwrap();
+
+			assert_eq!(
+				prev_balance - Balances::free_balance(2),
```

Comment:
Is `prev_balance - Balances::free_balance(2)` zero? It would be awesome to have the same comment as other test.

Reply:
No. Why should it be zero? I added an additional assert that tests for a fixed value.

Reply:
Thanks, I was just confused for a moment.